ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,137 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Example of setInterval
+var init_interval
+
+// Record timer waiting to run
+var exist_timer = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const removeTimer = (key, DB) => {
+    try {
+        //Delete thread when thread exists
+        if (exist_timer[key] != undefined ) {
+            clearTimeout(exist_timer[key].timer)
+        }
+        delete exist_timer[key];
```

Comment:
@line23, How about exist_timer[key] id undefined?

Reply:
during restore, if the planned delete time is more than 12 hours, the timer will be undefined, that is, no deletion thread will be added

Reply:
So why check @line 20? 
I'm curious why check undefined only for line21 but not line 23.

Reply:
line 23 is delete whole exist_timer, line 21 just deletes the timer created by setTimeout()